# Profitable Apps in Apple Store & Google Play Store

The purpose of this project is to identify profitable apps for an app development company. The company builds apps that are free to download and focuses on in app purchases as their main source of revenue.

In this project we will be observing apps in the Apple Store and Google Play Store. We will analyze the data sets to identify types of apps that attract a large user base for potential profitable margins. This is will give the app development company insights for their next app ideas.

---
As of September 2018, there were approximately 2 million iOS apps available on the App Store, and 2.1 million Android apps on Google Play Store.

Instead of collecting data from over 4 million apps, we will be using a sample of the data.

- A [data set](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps/home) containing data about approximately 7,000 iOS apps from the App Store; the data was collected in July 2017.
- A [data set](https://www.kaggle.com/lava18/google-play-store-apps/home) containing data about approximately 10,000 Android apps from Google Play; the data was collected in August 2018.

In [1]:
from csv import reader

#Apple Store apps data set

open_file = open('AppleStore.csv',encoding = "utf-8")
read_file = reader(open_file)
apple_data = list(read_file)
apple_data_header = apple_data[0]
apple_data = apple_data[1:]

#Google Play Store data set

open_file = open('googleplaystore.csv',encoding = "utf-8")
read_file = reader(open_file)
android_data = list(read_file)
android_data_header = android_data[0]
android_data = android_data[1:]


---
A function named **explore_data()** is created to read data set rows repeatedly in a more readable way. The function also allows us to observe the number of rows and columns for the data set being examined.

In [2]:
def explore_data(dataset, start, end, rows_and_columns = False):
    dataset_slice = dataset[start:end]    
    for row in dataset_slice:
        print(row)
        print('\n') # adds a new (empty) line after each row

    if rows_and_columns:
        print('Number of rows:', len(dataset))
        print('Number of columns:', len(dataset[0]))

---
Apple Store data set is being examined using the **explore_data()** function below.

In [3]:
print(apple_data_header)
print('\n')
explore_data(apple_data,0,5,True)

['', 'id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['1', '281656475', 'PAC-MAN Premium', '100788224', 'USD', '3.99', '21292', '26', '4', '4.5', '6.3.5', '4+', 'Games', '38', '5', '10', '1']


['2', '281796108', 'Evernote - stay organized', '158578688', 'USD', '0', '161065', '26', '4', '3.5', '8.2.2', '4+', 'Productivity', '37', '5', '23', '1']


['3', '281940292', 'WeatherBug - Local Weather, Radar, Maps, Alerts', '100524032', 'USD', '0', '188583', '2822', '3.5', '4.5', '5.0.0', '4+', 'Weather', '37', '5', '3', '1']


['4', '282614216', 'eBay: Best App to Buy, Sell, Save! Online Shopping', '128512000', 'USD', '0', '262241', '649', '4', '4.5', '5.10.0', '12+', 'Shopping', '37', '5', '9', '1']


['5', '282935706', 'Bible', '92774400', 'USD', '0', '985920', '5320', '4.5', '5', '7.5.1', '4+', 'Reference', '37'

There are 7,197 apps and 17 columns/categories in the Apple Store data set.

The most useful columns are: *'price'*, *'rating_count_tot'*, *'user_rating'*, *'cont_rating'*, & *'prime_genre'*

---
Google Play Store data set is being examined using the **explore_data()** function below.

In [4]:
print(android_data_header)
print('\n')
explore_data(android_data,0,5,True)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Eve

There are 10,841 apps and 13 columns/categories in the Google Play Store data set. 

The most useful columns are: *'App'*, *'Category'*, *'Rating'*, *'Installs'*, *'Type'*, & *'Price'*

---
## Data Cleaning

**Cleaning Incorrect Data**

According to the [discussion](https://www.kaggle.com/lava18/google-play-store-apps/discussion/66015) section for Google Play Store data, there is an error in row 10472. Since it is unknown if the analyst discussing the issue has removed the header for the data set or not, we will be checking row 10471 & 10472 in order to investigate the issue.

In [5]:
print(android_data_header)
print('\n')
print(android_data[10471])
print('\n')
print(android_data[10472])

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Xposed Wi-Fi-Pwd', 'PERSONALIZATION', '3.5', '1042', '404k', '100,000+', 'Free', '0', 'Everyone', 'Personalization', 'August 5, 2014', '3.0.0', '4.0.3 and up']


['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']


In row 10472, there are only 12 column entries, with the *'Category'* information missing in the 2nd column. This shifts all the other coulmns one to the left.  We will delete this row to avoid errors in the upcoming analysis.

In [6]:
del android_data[10472] #avoid running this more than once

---
**Cleaning Duplicate Data**

There are also duplicate data in the Google Play Store data set according to the [discussions](https://www.kaggle.com/lava18/google-play-store-apps/discussion/76316). 

In [7]:
individual_data = []
duplicate_data = []
for row in android_data:
    app_name = row[0]
    if app_name in individual_data:
        duplicate_data.append(app_name)
    elif app_name not in individual_data:
        individual_data.append(app_name)

In [8]:
print("# of duplicate apps:", len(duplicate_data))
print('\n')
print('Examples:',duplicate_data[:11])

# of duplicate apps: 1181


Examples: ['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings', 'Box', 'Zenefits', 'Google Ads', 'Google My Business', 'Slack', 'FreshBooks Classic']


In [9]:
print('Instagram' in individual_data)
print('Instagram' in duplicate_data)

True
True


In [10]:
print(android_data_header)
print('\n')
for row in android_data:
    name_app = row[0]
    if name_app == 'Instagram':
        print(row)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577446', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66509917', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']


As seen above, there are multiple duplicates of the same app. However, we won't be deleting duplicates randomly. If we observe the 'Reviews' category, the highest entry is '66577446'. We will choose this entry since it represents the latest data on the app.

In [11]:
expected_data_length = len(android_data) - len(duplicate_data)
print('Expected Google Play Data Set Length: ', expected_data_length)

Expected Google Play Data Set Length:  9659


In [12]:
reviews_max = {} #each app with max reviews 

for row in android_data:
    name = row[0]
    n_reviews = float(row[3])
    if name in reviews_max and reviews_max[name] < n_reviews:
        reviews_max[name] = n_reviews
    elif name not in reviews_max:
        reviews_max[name] = n_reviews

In [13]:
print(expected_data_length == len(reviews_max)) #proceed if TRUE, otherwise review code & dictionary

True


We will now use "reviews_max" dictionary to clean the android data set of the duplicate apps and create a new data set.

In [14]:
android_clean = [] #new data set without duplicates
already_added = [] #this data set helps to avoid duplicate apps with same number of reviews

for row in android_data:
    name = row[0]
    n_reviews = float(row[3])
    if (n_reviews == reviews_max[name]) and (name not in already_added):
        android_clean.append(row)
        already_added.append(name)

In [15]:
print(len(android_clean))
print('\n')
print('Expected data length equal to Cleaned Data Set Length: ', expected_data_length == len(android_clean))

9659


Expected data length equal to Cleaned Data Set Length:  True


The data set "android_clean" contains the new data set free of duplicate apps. This is verified as the length of the "android_clean" is equal to the expected length. More obervation on the new data set below.

In [16]:
print(android_data_header)
print('\n')
print(explore_data(android_clean,0,3,True))

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 9659
Number of columns: 13
None


---
**Cleaning non English Data**

There are non English app data in the "android_clean" data set. The company we are analyzing the data for is focusing on an English speaking audience. Therefore, we do not need these non English app data and we will remove them. Few examples below:

In [17]:

print(apple_data[441][2]) #app name in 3rd column for apple store data set
print(apple_data[6580][2])
print('\n')
print(android_clean[4412][0]) #app name in 1st column for google play store data set
print(android_clean[7940][0])

恵方コンパス.
【はらぺこパズル】ごはんに恋をした


中国語 AQリスニング
لعبة تقدر تربح DZ


All these characters that are specific to English texts are encoded using the ASCII standard. Each ASCII character has a corresponding number between 0 and 127 associated with it, and we can take advantage of that to build a function that checks an app name and tells us whether it contains non-ASCII characters.

We will use this to create a function to sort out the non English characters and hence the non English apps. 

In [18]:
def char_def(string):
    for character in string:
        if ord(character) > 127:
            return False
        
    return True

#testing the char_def function
print(char_def('Instagram'))
print(char_def('爱奇艺PPS -《欢乐颂2》电视剧热播'))
print(char_def('Docs To Go™ Free Office Suite'))
print(char_def('Instachat 😜'))

True
False
False
False


In [19]:
print(ord('™'))
print(ord('😜'))

8482
128540


Currently, the function to check for non English apps will remove some English apps with use other symbols in thier names such as ™, — , –, etc. These fall outside the ASCII range. 
To avoid loss of useful data, we will update the function to only remove an app if the name has more than 3 non English characters.

In [20]:
def char_def(string):
    non_eng = 0
    for character in string:
        if ord(character) > 127:
            non_eng += 1
        
    if non_eng > 3:
            print(non_eng)
            return False
    
    else:
            print(non_eng)
            return True

print(char_def('Docs To Go™ Free Office Suite'))
print(char_def('Instachat 😜'))
print(char_def('爱奇艺PPS -《欢乐颂2》电视剧热播'))

1
True
1
True
13
False


The function might still remove English apps with over 3 non_ASCII characters but the loss of data will be minimal.

Below we use the updated "char_def()" function to clean the data sets of non English apps.

In [21]:
apple_english = []

for row in apple_data:
    name = row[2]
    if char_def(name):
        apple_english.append(row)

        

0
0
0
0
0
0
0
0
0
0
0
0
1
0
0
0
0
0
0
0
0
0
0
0
0
1
0
0
0
0
0
0
0
0
0
1
1
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
1
0
0
0
0
0
0
0
1
3
0
18
0
0
0
0
0
0
0
0
0
1
0
1
0
8
0
0
0
0
0
0
0
0
0
1
0
0
0
0
0
0
0
0
0
0
1
0
0
0
0
1
0
0
0
0
0
0
1
0
0
0
0
0
0
1
0
1
0
0
0
0
0
0
0
1
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
1
0
0
0
0
0
0
0
0
0
0
0
0
1
0
0
1
0
0
0
0
0
0
0
0
0
1
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
1
0
0
0
1
0
0
0
0
0
14
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
1
0
0
0
0
0
1
4
0
0
0
0
0
0
0
0
2
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
16
0
0
0
0
0
0
0
0
0
1
0
0
0
0
0
0
0
0
0
0
0
0
0
1
0
0
0
0
0
0
0
0
0
0
0
2
1
0
0
0
0
0
10
3
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
1
0
0
0
0
0
0
0
1
0
0
22
0
0
0
0
0
0
0
0
0
0
0
1
0
0
0
0
0
0
0
0
1
0
0
0
0
0
0
0
0
0
0
0
1
0
0
1
0
0
0
0
0
0
0
0
0
0
0
0
0
0
1
0
0
1
0
0
0
0
0
0
0
0
0
0
0
0
0
0
1
16
0
1
0
0
0
0
1
0
0
0
0
0
0
0
17
0
0
3
0
0
0
0
0
0
0
0
0
0
0
2
0
1
0
1
0
0
4
6
0
0
0
0
0
0
1
0
0
0
14
0
17
1
0
0
0
0
0
0
0
0
0
0
1
1
8
0
0
0
14
12
0
0
0
1
0
0
0
0
0
0
0
17
0
18
0
1
0
0
0
0

0
0
0
0
0
0
1
0
0
0
1
0
0
0
0
0
0
0
0
0
0
19
0
17
0
0
0
18
12
7
8
22
35
0
0
0
0
0
20
0
0
1
0
0
0
0
0
0
0
0
0
0
0
0
0
0
20
0
0
0
0
1
0
0
0
0
0
0
0
0
0
0
0
19
1
1
0
0
0
0
0
8
0
1
19
0
0
0
0
0
0
0
0
0
0
0
12
13
0
16
0
0
28
26
0
0
0
0
0
0
1
8
0
0
1
0
0
0
12
0
0
17
0
0
0
17
0
0
0
0
0
0
16
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
13
0
0
0
0
0
0
0
0
11
0
0
0
0
0
0
0
11
9
0
0
0
0
0
0
0
0
6
0
0
5
18
0
0
0
1
0
0
0
0
0
0
0
0
0
0
0
0
0
7
0
0
0
0
0
0
0
0
14
0
0
0
0
0
1
1
3
11
0
0
0
0
0
0
0
15
0
0
0
0
24
1
0
9
0
0
0
1
8
0
0
0
0
0
0
0
0
31
0
0
8
0
0
0
1
0
21
0
0
0
0
0
0
0
0
0
0
0
14
0
0
0
22
0
2
1
0
0
0
0
0
0
1
0
0
0
0
0
0
3
0
0
0
0
0
0
0
0
0
0
0
11
0
0
0
0
0
0
0
10
0
0
0
0
0
0
0
0
0
20
0
0
0
0
0
0
0
0
0
0
0
0
0
33
0
0
1
0
0
0
0
4
0
0
0
0
0
0
0
1
1
0
16
0
0
1
0
24
0
0
0
0
0
0
0
0
16
0
0
0
0
0
0
0
0
0
28
0
0
4
0
10
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
10
20
0
0
0
11
29
0
22
0
0
0
31
0
1
0
0
0
0
0
4
0
0
1
1
0
0
0
0
0
14
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
8
0
0
0
0
0
0
0
22
0
0
0
0
6
0
0
0
1
29
0
0
8
0
7
0
26


In [22]:
android_english = []

for row in android_clean:
    name = row[0]
    if char_def(name):
        android_english.append(row)

0
1
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
1
0
0
1
2
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
1
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
1
0
0
1
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
1
0
0
0
0
0
0
0
0
0
0
0
0
1
0
0
1
0
0
0
0
0
3
2
0
2
0
0
0
1
0
0
2
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
1
0
0
0
0
0
0
1
0
0
0
0
0
0
0
0
0
0
0
0
1
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
1
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
1
0
0
0
0
1
0
0
0
0
0
0
0
0
0
0
1
0
0
0
0
0
1
0
0
0
0
0
0
0
0
0
0
0
0
1
0
0
0
0
0
0
0
0
1
0
1
0
0
0
0
0
0
0
0
0
1
0
0
0
0
0
0
0
0
0
0
0
0
1
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
1
0
0
0
0
0
0
1
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0


0
0
0
0
0
0
0
3
0
0
0
0
2
0
0
0
0
0
1
1
0
0
0
4
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
1
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
1
0
2
0
0
0
2
0
0
5
0
1
1
0
0
0
0
0
0
0
0
2
0
0
0
0
0
0
0
0
0
0
0
1
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
1
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
13
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
1
0
0
0
0
0
0
1
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
1
0
0
0
0
0
0
0
0
0
0
0
0
0
1
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
7
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0

0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
1
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
1
0
0
0
0
0
0
0
0
0
2
1
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
1
2
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
1
0
0
0
0
0
0
0
0
0
2
0
2
0
0
0
1
1
0
0
0
0
0
0
0
0
1
0
0
0
0
0
0
0
0
1
0
0
0
0
0
2
2
0
2
2
0
0
0
0
1
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
1
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
1
0
0
0
1
0
0
0
0
1
0
0
0
0
0
0
0
0
0
0
0
0
0
0
1
0
0
0
0
0
0
0
0
0
0
0
0
0
0
1
0
0
0
0
0
0
0
0
0
0
0
0
0
3
0
0
0
0
0
0
0
1
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
2
0
0
0
0
0
15
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
1
0
0
0
0
1
0
0
0
0
0
0
0
0
1
0
0
0
0
0
0
0
0
0
0
0
0
0
0

In [23]:
print("Apple English Data")
print(apple_data_header)
print(explore_data(apple_english,0,3,True))
print("\n")
print("Android English Data")
print(android_data_header)
print(explore_data(android_english,0,3,True))

Apple English Data
['', 'id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']
['1', '281656475', 'PAC-MAN Premium', '100788224', 'USD', '3.99', '21292', '26', '4', '4.5', '6.3.5', '4+', 'Games', '38', '5', '10', '1']


['2', '281796108', 'Evernote - stay organized', '158578688', 'USD', '0', '161065', '26', '4', '3.5', '8.2.2', '4+', 'Productivity', '37', '5', '23', '1']


['3', '281940292', 'WeatherBug - Local Weather, Radar, Maps, Alerts', '100524032', 'USD', '0', '188583', '2822', '3.5', '4.5', '5.0.0', '4+', 'Weather', '37', '5', '3', '1']


Number of rows: 6183
Number of columns: 17
None


Android English Data
['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']
['Photo Editor & Candy Camera & Grid & ScrapBook', 'A

There are 6,183 english apps in Apple data set & 9,614 english apps in Android data set.

---
## Isolate the Free Apps

As mentioned above, the company, the analysis is created for, is focused on apps that are free. We will be condensing the data sets further to work with only the free apps.

In [24]:
android_free = []
apple_free = []

for row in apple_english:
    price = float(row[5])
    if price == 0:
        apple_free.append(row)

for row in android_english:
    price = row[7]
    if price == "0":
        android_free.append(row)
        
print("Final numbers of Apple apps: ",len(apple_free))
print("Final Numbers of Android apps: ",len(android_free))

Final numbers of Apple apps:  3222
Final Numbers of Android apps:  8864


There are 3,222 free Apple Store apps and 8,864 free Android Play Store apps. We will use these data sets to conduct our analysis.

---
## Most Common Genres in Each Market

As we mentioned in the introduction, our aim is to determine the kinds of apps that are likely to attract more users because our revenue is highly influenced by the number of people using our apps.

To minimize risks and overhead, our validation strategy for an app idea is comprised of three steps:

Build a minimal Android version of the app, and add it to Google Play.
If the app has a good response from users, we develop it further.
If the app is profitable after six months, we build an iOS version of the app and add it to the App Store.

Because our end goal is to add the app on both Google Play and the App Store, we need to find app profiles that are successful on both markets. For instance, a profile that works well for both markets might be a productivity app that makes use of gamification.

---
We will be creating functions to analyze the "prime_genre" column Apple data set. For the Android data set, we will be analyzing the "Category" & "Genres" columns. 

In [25]:
def freq_table(dataset,index):
    table = {}
    total = 0
    for row in dataset:
        total +=1
        column = row[index]
        if column in table:
            table[column] += 1
        else:
            table[column] = 1
            
    percentages = {}
    for key in table:
        percent = (table[key]/total)*100
        percentages[key] = percent      
        
    return percentages

In [26]:
print(freq_table(apple_free,-5)) #index:-5 is "prime_genre" column in Apple data set.


{'Productivity': 1.7380509000620732, 'Weather': 0.8690254500310366, 'Shopping': 2.60707635009311, 'Reference': 0.5586592178770949, 'Finance': 1.1173184357541899, 'Music': 2.0484171322160147, 'Utilities': 2.5139664804469275, 'Travel': 1.2414649286157666, 'Social Networking': 3.2898820608317814, 'Sports': 2.1415270018621975, 'Health & Fitness': 2.0173805090006205, 'Games': 58.16263190564867, 'Food & Drink': 0.8069522036002483, 'News': 1.3345747982619491, 'Book': 0.4345127250155183, 'Photo & Video': 4.9658597144630665, 'Entertainment': 7.883302296710118, 'Business': 0.5276225946617008, 'Lifestyle': 1.5828677839851024, 'Education': 3.662321539416512, 'Navigation': 0.186219739292365, 'Medical': 0.186219739292365, 'Catalogs': 0.12414649286157665}


For readability we will create the function "display_function".

In [27]:
def display_table(dataset, index):
    table = freq_table(dataset, index)
    table_display = []
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)

    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], ':', entry[0])

In [28]:
display_table(apple_free,-5) #index:-5 is "prime_genre" column in Apple data set.

Games : 58.16263190564867
Entertainment : 7.883302296710118
Photo & Video : 4.9658597144630665
Education : 3.662321539416512
Social Networking : 3.2898820608317814
Shopping : 2.60707635009311
Utilities : 2.5139664804469275
Sports : 2.1415270018621975
Music : 2.0484171322160147
Health & Fitness : 2.0173805090006205
Productivity : 1.7380509000620732
Lifestyle : 1.5828677839851024
News : 1.3345747982619491
Travel : 1.2414649286157666
Finance : 1.1173184357541899
Weather : 0.8690254500310366
Food & Drink : 0.8069522036002483
Reference : 0.5586592178770949
Business : 0.5276225946617008
Book : 0.4345127250155183
Navigation : 0.186219739292365
Medical : 0.186219739292365
Catalogs : 0.12414649286157665


For the table above, we can observe that Games apps share 58.16% of the English, free apps in the Apple data set. Entertainment(7.88%), Photo % Video(~5%), and Education(3.66%) follows after, but with a huge gap from games. 
From this, we can deduct that when it comes to English, free apps in the Apple Store, apps made for games and entertainment occupy over 60% of the Apple Store.

---

In [29]:
display_table(android_free,1) # index:1 is the "Category" column in the Android Play Store data set.

FAMILY : 18.907942238267147
GAME : 9.724729241877256
TOOLS : 8.461191335740072
BUSINESS : 4.591606498194946
LIFESTYLE : 3.9034296028880866
PRODUCTIVITY : 3.892148014440433
FINANCE : 3.7003610108303246
MEDICAL : 3.531137184115524
SPORTS : 3.395758122743682
PERSONALIZATION : 3.3167870036101084
COMMUNICATION : 3.2378158844765346
HEALTH_AND_FITNESS : 3.0798736462093865
PHOTOGRAPHY : 2.944494584837545
NEWS_AND_MAGAZINES : 2.7978339350180503
SOCIAL : 2.6624548736462095
TRAVEL_AND_LOCAL : 2.33528880866426
SHOPPING : 2.2450361010830324
BOOKS_AND_REFERENCE : 2.1435018050541514
DATING : 1.861462093862816
VIDEO_PLAYERS : 1.7937725631768955
MAPS_AND_NAVIGATION : 1.3989169675090252
FOOD_AND_DRINK : 1.2409747292418771
EDUCATION : 1.1620036101083033
ENTERTAINMENT : 0.9589350180505415
LIBRARIES_AND_DEMO : 0.9363718411552346
AUTO_AND_VEHICLES : 0.9250902527075812
HOUSE_AND_HOME : 0.8235559566787004
WEATHER : 0.8009927797833934
EVENTS : 0.7107400722021661
PARENTING : 0.6543321299638989
ART_AND_DESIGN : 

The distribution of apps differ vastly among the English, free apps in the Google Play Store compared to the Apple Store. There is no one dominating category/ genre in the Google Play Store. The top category is Family with 18.91% with Game(9.72%), Tools(8.46%), and Business(4.59%) following after. There are more apps meant for practical usage(Finance, Business, Tools, etc) that are at the top of the list here. We will investigate further by observing the "Genres" column in the android data set.


In [30]:
display_table(android_free,-4) #index:-4 is "Genres" column in the Android data set

Tools : 8.449909747292418
Entertainment : 6.069494584837545
Education : 5.347472924187725
Business : 4.591606498194946
Productivity : 3.892148014440433
Lifestyle : 3.892148014440433
Finance : 3.7003610108303246
Medical : 3.531137184115524
Sports : 3.463447653429603
Personalization : 3.3167870036101084
Communication : 3.2378158844765346
Action : 3.1024368231046933
Health & Fitness : 3.0798736462093865
Photography : 2.944494584837545
News & Magazines : 2.7978339350180503
Social : 2.6624548736462095
Travel & Local : 2.3240072202166067
Shopping : 2.2450361010830324
Books & Reference : 2.1435018050541514
Simulation : 2.0419675090252705
Dating : 1.861462093862816
Arcade : 1.8501805054151623
Video Players & Editors : 1.7712093862815883
Casual : 1.7599277978339352
Maps & Navigation : 1.3989169675090252
Food & Drink : 1.2409747292418771
Puzzle : 1.128158844765343
Racing : 0.9927797833935018
Role Playing : 0.9363718411552346
Libraries & Demo : 0.9363718411552346
Auto & Vehicles : 0.9250902527075

The "Genres" column is more detailed to how the apps are categorized. Since we are focused on the overview of the app categorization, we will use the "Category" column moving forward.

According to the genre/ category column analysis, it seems apps designed for games and entertainment heavily dominate the Apple Store. Google Play Store is a mixture of apps designed for fun and apps designed for practical purposes.

---
## Most Popular Apps by Genre

**Apple Store Popular Genres**

One way to find out what genres are the most popular (have the most users) is to calculate the average number of installs for each app genre. For the Google Play data set, we can find this information in the "Installs" column, but this information is missing for the App Store data set. As a workaround, we'll take the total number of user ratings as a proxy, which we can find in the "rating_count_tot" app.

We created a function to observe the average number of user ratings per app genre on the Apple Store.

In [38]:
apple_genre = freq_table(apple_free,-5)

for genre_key in apple_genre:
    total = 0
    len_genre = 0
    for row in apple_free:
        genre_name = row[-5]
        if genre_name == genre_key:
            rating_tot = float(row[6])
            total += rating_tot
            len_genre += 1
    average = total / len_genre
    print(genre_key,":",average)

Productivity : 21028.410714285714
Weather : 52279.892857142855
Shopping : 26919.690476190477
Reference : 74942.11111111111
Finance : 31467.944444444445
Music : 57326.530303030304
Utilities : 18684.456790123455
Travel : 28243.8
Social Networking : 71548.34905660378
Sports : 23008.898550724636
Health & Fitness : 23298.015384615384
Games : 22788.6696905016
Food & Drink : 33333.92307692308
News : 21248.023255813954
Book : 39758.5
Photo & Video : 28441.54375
Entertainment : 14029.830708661417
Business : 7491.117647058823
Lifestyle : 16485.764705882353
Education : 7003.983050847458
Navigation : 86090.33333333333
Medical : 612.0
Catalogs : 4004.0


The categories with the most ratings, and therefore downloads/usages are Navigation, Reference and Social Networking. Lets investigate further to see the distribution of ratings within the categories/genres.

In [48]:
def genre_by_name(string):
    for row in apple_free:
        name = row[2]
        ratings = row[6]
        genre = row[-5]
        if genre == string:
            print(name,":",ratings) 
        

In [50]:
genre_by_name("Navigation")

Waze - GPS Navigation, Maps & Real-time Traffic : 345046
Geocaching® : 12811
ImmobilienScout24: Real Estate Search in Germany : 187
Railway Route Search : 5
CoPilot GPS – Car Navigation & Offline Maps : 3582
Google Maps - Navigation & Transit : 154911


In [51]:
genre_by_name("Reference")

Bible : 985920
Dictionary.com Dictionary & Thesaurus : 200047
Dictionary.com Dictionary & Thesaurus for iPad : 54175
Muslim Pro: Ramadan 2017 Prayer Times, Azan, Quran : 18418
Merriam-Webster Dictionary : 16849
Google Translate : 26786
Night Sky : 12122
WWDC : 762
Jishokun-Japanese English Dictionary & Translator : 0
教えて!goo : 0
VPN Express : 14
New Furniture Mods - Pocket Wiki & Game Tools for Minecraft PC Edition : 17588
LUCKY BLOCK MOD ™ for Minecraft PC Edition - The Best Pocket Wiki & Mods Installer Tools : 4693
Guides for Pokémon GO - Pokemon GO News and Cheats : 826
Horror Maps for Minecraft PE - Download The Scariest Maps for Minecraft Pocket Edition (MCPE) Free : 718
City Maps for Minecraft PE - The Best Maps for Minecraft Pocket Edition (MCPE) : 8535
GUNS MODS for Minecraft PC Edition - Mods Tools : 1497
Real Bike Traffic Rider Virtual Reality Glasses : 8


In [52]:
genre_by_name("Social Networking")

Facebook : 2974676
LinkedIn : 71856
Skype for iPhone : 373519
Tumblr : 334293
Match™ - #1 Dating App. : 60659
WhatsApp Messenger : 287589
TextNow - Unlimited Text + Calls : 164963
Grindr - Gay and same sex guys chat, meet and date : 23201
imo video calls and chat : 18841
Ameba : 269
Weibo : 7265
Badoo - Meet New People, Chat, Socialize. : 34428
Kik : 260965
Qzone : 1649
Fake-A-Location Free ™ : 354
Tango - Free Video Call, Voice and Chat : 75412
MeetMe - Chat and Meet New People : 97072
SimSimi : 23530
Viber Messenger – Text & Call : 164249
Find My Family, Friends & iPhone - Life360 Locator : 43877
Weibo HD : 16772
POF - Best Dating App for Conversations : 52642
GroupMe : 28260
Lobi : 36
WeChat : 34584
ooVoo – Free Video Call, Text and Voice : 177501
Pinterest : 1061624
知乎 : 397
Qzone HD : 458
Skype for iPad : 60163
LINE : 11437
QQ : 9109
LOVOO - Dating Chat : 1985
QQ HD : 5058
Messenger : 351466
eHarmony™ Dating App - Meet Singles : 11124
YouNow: Live Stream Video Chat : 12079
Cougar 

For "Navigation" & "Social Networking", the ratings in the categories are heavily influences by a few apps. For "Navigation" its Waze & Google Maps & for "Social Networking" its apps such as Facebook, Skype & ooVoo.

For "References", the distribution is wider and has potential room for apps in the future. Apps that are references to Navigation & Social Networking can created to aid in help with utilzing the most common apps in those categories. Another idea would be to create an audiobook or ebook in different languages for English speakers.

The other categories(Weather, Food & Drink, Finance) do not have as much popularity and therefore low potential profitabilty. Unless the next app idea is revolutionary, pursuing ideas in those categories won't be beneficial.

---
**Google Play Store Popular Genres**

We have data about the number of installs for the Google Play market, so we should be able to get a clearer picture about genre popularity. However, the install numbers don't seem precise enough — we can see that most values are open-ended (100+, 1,000+, 5,000+, etc.)

In [53]:
display_table(android_free,5)

1,000,000+ : 15.726534296028879
100,000+ : 11.552346570397113
10,000,000+ : 10.548285198555957
10,000+ : 10.198555956678701
1,000+ : 8.393501805054152
100+ : 6.915613718411552
5,000,000+ : 6.825361010830325
500,000+ : 5.561823104693141
50,000+ : 4.7721119133574
5,000+ : 4.512635379061372
10+ : 3.5424187725631766
500+ : 3.2490974729241873
50,000,000+ : 2.3014440433213
100,000,000+ : 2.1322202166064983
50+ : 1.917870036101083
5+ : 0.78971119133574
1+ : 0.5076714801444043
500,000,000+ : 0.2707581227436823
1,000,000,000+ : 0.22563176895306858
0+ : 0.04512635379061372
0 : 0.01128158844765343


One problem with this data is that is not precise. For instance, we don't know whether an app with 100,000+ installs has 100,000 installs, 200,000, or 350,000. However, we don't need very precise data for our purposes — we only want to get an idea which app genres attract the most users, and we don't need perfect precision with respect to the number of users.

We're going to leave the numbers as they are, which means that we'll consider that an app with 100,000+ installs has 100,000 installs, and an app with 1,000,000+ installs has 1,000,000 installs, and so on.

To perform computations, however, we'll need to convert each install number to float — this means that we need to remove the commas and the plus characters, otherwise the conversion will fail and raise an error. We'll do this directly in the loop below, where we also compute the average number of installs for each genre (category).

In [58]:
category_dict = freq_table(android_free,1)

for category in category_dict:
    total = 0
    len_category = 0
    for row in android_free:
        category_app = row[1]
        if category_app == category:
            installs = row[5]
            installs = installs.replace('+','')
            installs = installs.replace(',','')
            installs = float(installs)
            total += installs
            len_category += 1
    average = total / len_category
    print(category,":",average)
        

ART_AND_DESIGN : 1986335.0877192982
AUTO_AND_VEHICLES : 647317.8170731707
BEAUTY : 513151.88679245283
BOOKS_AND_REFERENCE : 8767811.894736841
BUSINESS : 1712290.1474201474
COMICS : 817657.2727272727
COMMUNICATION : 38456119.167247385
DATING : 854028.8303030303
EDUCATION : 1833495.145631068
ENTERTAINMENT : 11640705.88235294
EVENTS : 253542.22222222222
FINANCE : 1387692.475609756
FOOD_AND_DRINK : 1924897.7363636363
HEALTH_AND_FITNESS : 4188821.9853479853
HOUSE_AND_HOME : 1331540.5616438356
LIBRARIES_AND_DEMO : 638503.734939759
LIFESTYLE : 1437816.2687861272
GAME : 15588015.603248259
FAMILY : 3695641.8198090694
MEDICAL : 120550.61980830671
SOCIAL : 23253652.127118643
SHOPPING : 7036877.311557789
PHOTOGRAPHY : 17840110.40229885
SPORTS : 3638640.1428571427
TRAVEL_AND_LOCAL : 13984077.710144928
TOOLS : 10801391.298666667
PERSONALIZATION : 5201482.6122448975
PRODUCTIVITY : 16787331.344927534
PARENTING : 542603.6206896552
WEATHER : 5074486.197183099
VIDEO_PLAYERS : 24727872.452830188
NEWS_AND_

In [73]:
def dom_category_apps(string):
    for row in android_free:
        name = row[0]
        category = row[1]
        installs = row[5]
        if category == string and (installs == '1,000,000,000+' or installs == '500,000,000+' or installs == '100,000,000+'):
            print(name, ':', installs)

In [74]:
dom_category_apps('COMMUNICATION')

WhatsApp Messenger : 1,000,000,000+
imo beta free calls and text : 100,000,000+
Android Messages : 100,000,000+
Google Duo - High Quality Video Calls : 500,000,000+
Messenger – Text and Video Chat for Free : 1,000,000,000+
imo free video calls and chat : 500,000,000+
Skype - free IM & video calls : 1,000,000,000+
Who : 100,000,000+
GO SMS Pro - Messenger, Free Themes, Emoji : 100,000,000+
LINE: Free Calls & Messages : 500,000,000+
Google Chrome: Fast & Secure : 1,000,000,000+
Firefox Browser fast & private : 100,000,000+
UC Browser - Fast Download Private & Secure : 500,000,000+
Gmail : 1,000,000,000+
Hangouts : 1,000,000,000+
Messenger Lite: Free Calls & Messages : 100,000,000+
Kik : 100,000,000+
KakaoTalk: Free Calls & Text : 100,000,000+
Opera Mini - fast web browser : 100,000,000+
Opera Browser: Fast and Secure : 100,000,000+
Telegram : 100,000,000+
Truecaller: Caller ID, SMS spam blocking & Dialer : 100,000,000+
UC Browser Mini -Tiny Fast Private & Secure : 100,000,000+
Viber Mess

With further research, we see that "Communication" category for Android Play Store is similar to the "Social Networkng" genre in the Apple Store. Just as the "Social Networking" genre, "Communications" is dominated by a few apps, such as WhatsApp, Skype, Google Chrome, etc. These apps heavily skews the distribution and shows the category is not as popular for new  apps as it seems on the surface.


In [75]:
dom_category_apps('VIDEO_PLAYERS')

YouTube : 1,000,000,000+
Motorola Gallery : 100,000,000+
VLC for Android : 100,000,000+
Google Play Movies & TV : 1,000,000,000+
MX Player : 500,000,000+
Dubsmash : 100,000,000+
VivaVideo - Video Editor & Photo Movie : 100,000,000+
VideoShow-Video Editor, Video Maker, Beauty Camera : 100,000,000+
Motorola FM Radio : 100,000,000+


"Video Players" is also dominated by a few established apps.

we will research "Books & References" since it showed some promise in the Apple data set & has almost 9 million installs in the Google data set.

In [76]:
for row in android_free:
    name = row[0]
    category = row[1]
    installs = row[5]
    if category == 'BOOKS_AND_REFERENCE':
        print(name, ':', installs)

E-Book Read - Read Book for free : 50,000+
Download free book with green book : 100,000+
Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Free Panda Radio Music : 100,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
English Grammar Complete Handbook : 500,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
Google Play Books : 1,000,000,000+
AlReader -any text book reader : 5,000,000+
Offline English Dictionary : 100,000+
Offline: English to Tagalog Dictionary : 500,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
Recipes of Prophetic Medicine for free : 500,000+
ReadEra – free ebook reader : 1,000,000+
Anonymous caller detection : 10,000+
Ebook Reader : 5,000,000+
Litnet - E-books : 100,000+
Read books online : 5,000,000+
English to Urdu Dictionary : 500,000+
eBoox: book reader fb2 epub zip : 1,000,000+
English Persian Dictionary : 500,000+
Flybook : 500,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
E

In [77]:
dom_category_apps('BOOKS_AND_REFERENCE')

Google Play Books : 1,000,000,000+
Bible : 100,000,000+
Amazon Kindle : 100,000,000+
Wattpad 📖 Free Books : 100,000,000+
Audiobooks from Audible : 100,000,000+


In [81]:
for row in android_free:
        name = row[0]
        category = row[1]
        installs = row[5]
        if category == "BOOKS_AND_REFERENCE" and (installs == '1,000,000+'
                                            or installs == '5,000,000+'
                                            or installs == '10,000,000+'
                                            or installs == '50,000,000+'):
            print(name, ':', installs)

Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
AlReader -any text book reader : 5,000,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
ReadEra – free ebook reader : 1,000,000+
Ebook Reader : 5,000,000+
Read books online : 5,000,000+
eBoox: book reader fb2 epub zip : 1,000,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
Moon+ Reader : 10,000,000+
English-Myanmar Dictionary : 1,000,000+
Golden Dictionary (EN-AR) : 1,000,000+
All Language Translator Free : 1,000,000+
Aldiko Book Reader : 10,000,000+
Dictionary - WordWeb : 5,000,000+
50000 Free eBooks & Free AudioBooks : 5,000,000+
Al-Quran (Free) : 10,000,000+
Al Quran Indonesia : 10,000,000+
Al'Quran Bahasa Indonesia : 10,000,000+
Al Quran Al karim : 1,000,000+
Al Quran : EAlim - Translations & MP3 Offline : 5,000,000+
Koran Read &MP3 30 Juz Offline : 1,000,000+
H

Even though there are a few dominating apps in the "Books & Reference" section, there is diversity and room for new apps in this category.
It would be beneficial for any new apps to special features unseen in the market to rise among the ranks. The competetion is not as brutal as the other categories with investigated previously.

---
## Conclusion

In this project, we analyzed data about the App Store and Google Play mobile apps with the goal of recommending an app profile that can be profitable for both markets.

Since we are deploying the app first in the Google Play Store, the chances of success are much higher if the app is created for a practical purpose. "Books & Reference" category shows the most promise.
An app that allows you to create text from audio or create audio from written text may be unique enough to rise the ranks among the category.
Once it is launched in the Apple Store, it would be beneficial to create for the app with gamification features (such as the user earns points through usage) since games are a popular in the Apple Store.